src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,102 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const handleChange = (name: string, value: string) => {
+    setNewMovie({
+      ...newMovie,
+      [name]: value,
+    });
+  };
+
+  const reset = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
```

Comment:
To avoid code duplicating you can create varaible with empty object and use it 
```
const EMPTY_MOVIE = {
    title: '',
    description: '',
    imgUrl: '',
    imdbUrl: '',
    imdbId: ''
};
```
```suggestion
  const [newMovie, setNewMovie] = useState(EMPTY_MOVIE );

  const handleChange = (name: string, value: string) => {
    setNewMovie({
      ...newMovie,
      [name]: value,
    });
  };

  const reset = () => {
    setNewMovie(EMPTY_MOVIE);
  };
```